src/jit/codegenarm.cpp

```diff
@@ -44,7 +44,39 @@ void CodeGen::genSetRegToIcon(regNumber reg, ssize_t val, var_types type, insFla
 //
 void CodeGen::genEmitGSCookieCheck(bool pushReg)
 {
-    NYI("ARM genEmitGSCookieCheck");
+    noway_assert(compiler->gsGlobalSecurityCookieAddr || compiler->gsGlobalSecurityCookieVal);
+
```

Comment:
as far as I can tell, this GS cookie check function is almost exactly the same as the one for arm64. Can we share them, and avoid having this code duplication?